src/main/scala/uk/gov/hmrc/ct/accounts/frs102/calculations/BalanceSheetDebtorsCalculator.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ct.accounts.frs102.calculations
+
+import uk.gov.hmrc.ct.accounts.frs102.boxes._
+
+trait BalanceSheetDebtorsCalculator extends DebitAwareCalculation  {
+
+  def calculateAC140(ac134: AC134, ac136: AC136, ac138: AC138): AC140 = {
+    (ac134.value, ac136.value, ac138.value) match {
+      case (None, None, None) => AC140(None)
+      case _ => sum(ac134, ac136, ac138)(AC140.apply)
```

Comment:
The sum function takes care of the None, None business, so you can delete the case match altogether